include/swift/SIL/TypeSubstCloner.h

```diff
@@ -82,9 +82,7 @@ class TypeSubstCloner : public SILClonerWithScopes<ImplClass> {
             // substitutions may be different from the original ones, e.g.
             // there can be less substitutions.
             RecursiveSubs = SubstitutionMap::get(
-              AI.getFunction()
-                ->getLoweredFunctionType()
-                ->getInvocationGenericSignature(),
+              LoweredFnTy->getSubstGenericSignature(),
```

Comment:
I still don't have a clear understanding of generic signature APIs. Since we're constructing an `apply` of this function type, I would have used `getInvocationGenericSignature()`. Why am I wrong? Is there some kind of rule I can remember about when to use `getSubstGenericSignature()`?

Reply:
This was a recent addition to the API. I think @jckarter is better equipped to explain this one.

Reply:
Yes, `getInvocationGenericSignature` is what you would use to bind the substitution map for an apply.

Reply:
@jckarter Thanks! I can remember that. Under what situations does `getSubstGenericSignature()` make sense?

Reply:
@atrick `getSubstGenericSignature` is the signature you use to apply the function type's own substitutions in order to get the concrete parameter and return types. For something like `<A, B, C> in (A, B) -> C for <Int, String, Float>`, the invocation generic signature will be empty (because the function type is fully concrete), and the subst generic signature will be `<A, B, C>`, because that's what you use to substitute the `<Int, String, Float>` arguments into the type to get the concrete parameter and return types.

In case you missed it, there's a writeup in this thread: https://forums.swift.org/t/improving-the-representation-of-polymorphic-interfaces-in-sil-with-substituted-function-types/29711